lib/repl.js

```diff
@@ -851,23 +851,28 @@ function complete(line, callback) {
           // Exclude versioned names that 'npm' installs.
           continue;
         }
-        if (exts.indexOf(ext) !== -1) {
+        abs = path.resolve(dir, name);
+        try {
+          isDirectory = fs.statSync(abs).isDirectory();
+        } catch (e) {
+          continue;
+        }
+        if (!isDirectory && exts.indexOf(ext) !== -1) {
           if (!subdir || base !== 'index') {
             group.push(subdir + base);
           }
-        } else {
-          abs = path.resolve(dir, name);
+        } else if (isDirectory) {
```

Comment:
Nit: you could get rid of one of the `isDirectory` checks (the churn is there anyway).
```js
if (isDirectory) {
  // ...
} else if (exts.includes(ext)) {
  // ...
}
```